addons/binding/org.openhab.binding.synopanalyzer/src/main/java/org/openhab/binding/synopanalyzer/handler/SynopAnalyzerHandler.java

```diff
@@ -61,6 +62,7 @@
 
     public SynopAnalyzerHandler(Thing thing) {
         super(thing);
+        SYNOP_DATE_FORMAT.setTimeZone(TimeZone.getTimeZone("UTC"));
```

Comment:
In my view, indeed the only reason for all of this is to have the time in UTC for the query to synop. Before using this binding, I did exactly this in a direct http query from a rule (see [here](https://community.openhab.org/t/virtual-solar-sensor/24705/7)). I made the minimal change possible to the binding code, although I also believe the Java 8 solution is better.
To be very clear, this code change actually fixes two distinct problems. One is the conversion to UTC, the second is the fact that Calendar.role does not update the higher level field if you run over the limit of 12 hours. It is the last one that causes the behaviour in the issue description. The first one goes more unnoticed, as the binding goes back up to 24 hours. It would typically hit a good value if east of Greenwich.
Going back max 24 hours is required as not all synop stations have an update every hour. I do think you can actually reduce it to 1 query, going 24 hours back. The first line in the response will always be the most recent update. One can discard the rest.
I am happy to do these changes, including changing it all to the Java 8 classes, but probably not within the next 3 weeks. If @martinvw or @clinique want to do something, please go ahead.